03-Modify-Raster.Rmd

```diff
@@ -74,197 +72,240 @@ Read more about the `raster` package in R.</a>
-objects on the earth. In comparison, a `Digital Terrain Model (DTM)` or `Digital Elevation
-Model (DEM)` contains elevation data for the actual ground (with trees, buildings and 
-other objects removed). 
-
-In ecology, we are often interested in measuring the heights of trees and so the
-raster that we really want is the `difference` between the `DSM` and the `DTM`.
-This data product is often referred to as a `Canopy Height Model (CHM)` and represents
-the actual height of trees, buildings etc above the ground.
-
-We can calculate this difference by subtracting the two rasters in `R`. However
-if our rasters get large, we can calculate the difference more efficiently using
-the `overlay` function. 
-
-First, let's create and plot a `CHM` by subtracting two rasters - we often call
-this `raster math` in the geospatial world.
-
+As introduced in Lesson 02, we often want to overlay two or more rasters on top 
+of each other, and perform calculations to create a new output raster. If 
+interested in measuring the heights of trees, then what we really want is the 
+`difference` between the Digital Surface Model (`DSM`, tops of trees) and the 
+Digital Terrain Model (`DTM`, ground level) .This data product is often referred
+to as a Canopy Height Model (`CHM`) and represents the actual height of trees, 
+buildings, etc above the ground.
+
+We can calculate this difference in two different ways:
+1) by subtracting the two rasters in `R` 
+or if our rasters are large
+2) using the `overlay()` function to more efficiently calculate the difference.
+
+First, we'll learn the intuitive first step of subtracting layers to create and
+plot a `CHM`.  In the geospatial world we often call
+this *raster math*.
+
+If you completed Lessons 00 and 02 you've already loaded and viewed 
+`HARV_dtmCrop.tif` and `HARV_dsmCrop.tif` and can skip this code chunk.  If you 
+have not completed it, let's load and check out our raster files. 
 ```{r load-libraries }
-
 #load raster package
 library(raster)
 
-#view info about the dtm raster data
+#view info about the dtm & dsm raster data
 GDALinfo("NEON_RemoteSensing/HARV/DTM/HARV_dtmCrop.tif")
-
-#load the DTM
+GDALinfo("NEON_RemoteSensing/HARV/DTM/HARV_dsmCrop.tif")
+#load the DTM & DSM rasters
 DTM <- raster("NEON_RemoteSensing/HARV/DTM/HARV_dtmCrop.tif")
-
-#create a quick plot
+DSM <- raster("NEON_RemoteSensing/HARV/DSM/HARV_dsmCrop.tif")
+#create a quick plot of each to see what we're dealing with
 plot(DTM,
      main="Digital Terrain Model (Elevation)\n Harvard Forest")
 
-#If it isn't already loaded, load the DSM as well 
-DSM <- raster("NEON_RemoteSensing/HARV/DSM/HARV_dsmCrop.tif")
-
+plot(DSM,
+     main="Digital Surface Model (Elevation)\n Harvard Forest")
 ```
 
 Once the data are loaded, we can use mathametical operations on them.
-For instance, we can subtract the DTM from the DSM to create a `Canopy Height
-Model`.
+For instance, we can subtract the DTM from the DSM to create a Canopy Height
+Model.
```

Comment:
I'm thinking that it's just a discussion of how it could be done but it would not get taught that way. So it could even be a short paragraph. Maybe we leave it for now?

Leah
--
Leah A. Wasser, Ph.D.
Supervising Scientist
Education / Data Products
National Ecological Observatory Network (NEON)
Boulder, Colorado
--------------------
For free tutorials and resources on working with spatio-temporal data:
http://www.neondataskills.org<http://www.neondataskills.org/>

From: "Megan A. Jones" <notifications@github.com<mailto:notifications@github.com>>
Reply-To: data-lessons/NEON-R-Spatial-Raster <reply@reply.github.com<mailto:reply@reply.github.com>>
Date: Monday, November 16, 2015 at 2:36 PM
To: data-lessons/NEON-R-Spatial-Raster <NEON-R-Spatial-Raster@noreply.github.com<mailto:NEON-R-Spatial-Raster@noreply.github.com>>
Cc: Leah Wasser <lwasser@neoninc.org<mailto:lwasser@neoninc.org>>
Subject: Re: [NEON-R-Spatial-Raster] Lessons 03 and 04 Rearrangements and Edits (#27)


In 03-Modify-Raster.Rmd<https://github.com/data-lessons/NEON-R-Spatial-Raster/pull/27#discussion_r44986992>:

>  ```
>
>  Once the data are loaded, we can use mathametical operations on them.
> -For instance, we can subtract the DTM from the DSM to create a `Canopy Height
> -Model`.
> +For instance, we can subtract the DTM from the DSM to create a Canopy Height
> +Model.


I'm okay with using it as an example but not teaching, however, I'm not sure how this would be done as it is currently 1 line of code, and plotting it to see the changes. Are you just thinking making that chunk echo=FALSE?

The Challenge is to check understanding of the concept. It could be moved after Overlay.

-
Reply to this email directly or view it on GitHub<https://github.com/data-lessons/NEON-R-Spatial-Raster/pull/27/files#r44986992>.